app/models/user.rb

```diff
@@ -27,6 +29,7 @@ class User < ApplicationRecord
   before_validation :geocode, if: ->(v) { v.zip.present? && v.zip_changed? }
   before_save :upcase_state
   before_save :downcase_email
+  after_commit :notify_leaders_on_geocode_update, on: [:create, :update]
```

Comment:
Since this is exclusive to both `:create` and `:update`, I believe you can just truncate this to just `on: :save`